src/plugins/dashboard_embeddable_container/public/embeddable/viewport/dashboard_viewport.tsx

```diff
@@ -82,24 +98,34 @@ export class DashboardViewport extends React.Component<DashboardViewportProps, S
     });
   };
 
+  private renderEmptyScreen() {
+    const emptyScreenProps = this.props.container.getInput().emptyScreenProps;
+    if (!emptyScreenProps) {
+      return null;
+    }
+    return <DashboardEmptyScreen {...emptyScreenProps} />;
+  }
+
   public render() {
     const { container } = this.props;
+    const { isFullScreenMode, isEmptyState, panels, title, description, useMargins } = this.state;
     return (
-      <div
-        data-shared-items-count={Object.values(this.state.panels).length}
-        data-shared-items-container
-        data-title={this.state.title}
-        data-description={this.state.description}
-        className={
-          this.state.useMargins ? 'dshDashboardViewport-withMargins' : 'dshDashboardViewport'
-        }
-      >
-        {this.state.isFullScreenMode && (
-          <this.context.services.ExitFullScreenButton
-            onExitFullScreenMode={this.onExitFullScreenMode}
-          />
-        )}
-        <DashboardGrid container={container} />
+      <div>
```

Comment:
Nit: Can this be replaced with a Fragment? It doesn't seem like you need the extra `div` wrapper for anything...

```suggestion
      <>
```

Reply:
I think the reporting didn't work properly if this wasn't a div, but will try again.

Reply:
yes, for some reason report generation fails if this is not wrapped in a `div`:
<img width="656" alt="Screenshot 2019-12-03 at 18 07 41" src="https://user-images.githubusercontent.com/1937956/70077032-e9b0b980-15f7-11ea-9eaf-e077c6dd3965.png">

Reply:
Ah, it works with the `Fragment`. But otherwise, this DOM structure is needed for reporting to be generated properly, as it needs to have an element with `data-shared-items-container` and this cannot be wrapped around empty screen, or the content of empty screen would be generated in the report as well. This will need to change when we introduce the empty panel, but I think it's good enough for now.